front/pages/poke/[wId]/index.tsx

```diff
@@ -0,0 +1,147 @@
+import { UserType, WorkspaceType } from "@app/types/user";
+
+export const getServerSideProps: GetServerSideProps<{
+  user: UserType;
+  workspace: WorkspaceType;
+}> = async (context) => {
+  const session = await getSession(context.req, context.res);
+  const user = await getUserFromSession(session);
+  const wId = context.params?.wId;
+
+  if (!user) {
+    return {
+      redirect: {
+        destination: "/login",
+        permanent: false,
+      },
+    };
+  }
+
+  if (!user.isDustSuperUser) {
+    return {
+      notFound: true,
+    };
+  }
+
+  if (!wId || typeof wId !== "string") {
+    return {
+      notFound: true,
+    };
+  }
+
+  const auth = await Authenticator.fromSession(session, wId);
+  const workspace = auth.workspace();
+
+  if (!workspace) {
+    return {
+      notFound: true,
+    };
+  }
+
+  return {
+    props: {
+      user,
+      workspace,
+    },
+  };
+};
+
+const WorkspacePage = ({
+  workspace,
+}: InferGetServerSidePropsType<typeof getServerSideProps>) => {
+  const router = useRouter();
+
+  const onUpgrade = async () => {
+    if (!window.confirm("Are you sure you want to upgrade this workspace?")) {
+      return;
+    }
+    try {
+      const r = await fetch(`/api/poke/workspaces/${workspace.sId}/upgrade`, {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+      });
+      if (!r.ok) {
+        throw new Error("Failed to upgrade workspace.");
+      }
+      await router.reload();
+    } catch (e) {
+      console.error(e);
+      window.alert("An error occurred while upgrading the workspace.");
+    }
+  };
+
+  const onDowngrade = async () => {
+    if (!window.confirm("Are you sure you want to downgrade this workspace?")) {
+      return;
+    }
+    try {
+      const r = await fetch(`/api/poke/workspaces/${workspace.sId}/downgrade`, {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+      });
+      if (!r.ok) {
+        throw new Error("Failed to downgrade workspace.");
+      }
+      await router.reload();
+    } catch (e) {
+      console.error(e);
+      window.alert("An error occurred while downgrading the workspace.");
+    }
+  };
+
+  const looksFullyUpgraded =
+    workspace.plan?.limits.dataSources.count === -1 &&
+    workspace.plan?.limits.dataSources.documents.count === -1 &&
+    workspace.plan?.limits.dataSources.documents.sizeMb === -1 &&
+    workspace.plan?.limits.dataSources.managed === true;
+
+  return (
+    <div className="min-h-screen bg-structure-50">
+      <PokeNavbar />
+      <div className="flex-grow p-6">
+        <>
+          <h1 className="mb-4 text-2xl font-bold">{workspace.name}</h1>
+          <h2 className="text-md mb-4 font-bold">Plan:</h2>
+          {looksFullyUpgraded ? (
+            <p className="mb-4 text-green-600">
+              This workspace looks fully upgraded.
```

Comment:
What would be "partially upgraded then?

NIT : "is" instead of  "looks"?

Reply:
> What would be "partially upgraded then?

If it had some of the limits bumped but not all... unclear how we'd end up in that state but 🤷 

> NIT : "is" instead of "looks"?

yeah, I'll switch it